baxter_collaboration_lib/include/robot_utils/utils.h

```diff
@@ -107,6 +107,16 @@ void setPosition(geometry_msgs::Pose& pose, float x, float y, float z);
  */
 void setOrientation(geometry_msgs::Pose& pose, float x, float y, float z, float w);
 
+/**
+ * replaces all instances of a substring with specified substring
+ * @param target      string containing the substring
+ * @param replaceThis substring to be replaced
+ * @param replaceWith substring to replace with
+ *
+ * return N/A
+ */
+void subStringReplace(std::string& target, std::string replaceThis, std::string replaceWith);
```

Comment:
Sure. I wanted to preserve the code that was already present that used stringstream. But the str() method with stringstream would not convert something like -0.0 to 0 (instead it would give -0). So for just the toString(double) situation, the fix was easy enough, but for the toString(doubleVector) and toString(Pose) situations, there was the possibility of multiple -0s being present in the final string. So I thought I'd create a new function in utils that I could use for both situations.

Within the toString functions, a vector like [-0.0, 1, 2, 3, -0] now becomes [0, 1, 2, 3, 0]. Same for a Pose. I tried to think of potential dangers of such a function, but in my use, I have only targeted "-0," and "-0}" substrings for replacement.